features/step_definitions/password_steps.rb

```diff
@@ -56,3 +56,28 @@ def visit_url_in_email(email, subject)
   click_on "Send instructions"
   page.should have_selector(RECAPTCHA_SCRIPT, visible: false)
 end
+
+Then "it should not be possible to reset their password" do
+  assert_not has_link?('Forgot password?', visible: :all)
+
+  expect do
+    visit path_to('the provider reset password page') # FIXME: first time does not raise??
+    visit path_to('the provider reset password page')
```

Comment:
This is certainly strange

Reply:
It's puzzling why a second open of the page is needed to see exceptions. But isn't this kind of test much better suited for integration tests? It doesn't seem to rely on javascript.

Moreover, if the test is not tagged with `@javascript` then it works fine with single opening of the page.

The issue might be related to using the chrome webdriver but more likely it has to do with the threading execution differences when `@javascript` is used.  The `@javascript` tag is handled by default by `rails-cucumber` but IIRC we override that because we chose to use the tag for browser selection.

Reply:
One more interesting find is from docs https://github.com/teamcapybara/capybara

> Server errors will only be raised in the session that initiates the server thread. If you are testing for specific server errors and using multiple sessions make sure to test for the errors using the initial session (usually :default)